### PR TITLE
ACT-1087 | typing for EntityType patterns, keywords and validation

### PR DESCRIPTION
**Pull request by bentatum**

### CODE REVIEW

File: src/sdk/models/entityTypes.ts [hunk 1]
```diff
@@ -1,6 +1,27 @@
...
+	VeryStrong = 'very_strong'
+}
+export interface EntityTypeRegexPattern {
+	description: string;
+	pattern: string;
+	priority: number;
+	confidence: ConfidenceLevel;
+}
+
+export interface EntityTypeKeyword {
```
Comment (JimPartinSkySync) [inline src/sdk/models/entityTypes.ts:19]: I have a branch with this and support for bringing in the actual keywords - https://github.com/portalarchitects/skysync-cli/commit/689f00a927c9be30f085dea2f94230f0fba5a7ad. Not a PR just yet but close. So two options:
1) Could you take out EntityTypeKeyword and keywords property below in EntityType and we will use the ones in my PR?
or 
2) We can merge what I need into this. But includes some additional files.

File: src/sdk/models/entityTypes.ts [hunk 2]
```diff
@@ -9,6 +30,9 @@ export interface EntityType extends IEntityIdentifier<string>, IAuditedEntity {
 	threshold?: number;
 	readonly?: boolean;
 	disabled?: boolean;
+	patterns?: EntityTypeRegexPattern;
+	keywords?: EntityTypeKeyword;
```
Comment (JimPartinSkySync) [inline src/sdk/models/entityTypes.ts:34]: See comment above.

File: src/sdk/models/entityTypes.ts (hunk 1 above)
Comment (bentatum) [reply src/sdk/models/entityTypes.ts]: Thanks Jim. I went ahead and removed it.

File: src/sdk/models/entityTypes.ts [hunk 3]
```diff
@@ -9,6 +24,7 @@ export interface EntityType extends IEntityIdentifier<string>, IAuditedEntity {
 	threshold?: number;
 	readonly?: boolean;
 	disabled?: boolean;
+	patterns?: EntityTypeRegexPattern;
```
Comment (smbecker) [inline src/sdk/models/entityTypes.ts:27]:
```suggestion
	patterns?: EntityTypeRegexPattern[];
```

**Review by smbecker — CHANGES_REQUESTED** [review]

File: src/sdk/models/entityTypes.ts [hunk 4]
```diff
@@ -1,6 +1,21 @@
...
 import { Category } from './categories';
 
+export enum ConfidenceLevel {
+	None = 'none',
+	VeryWeak = 'very_weak',
+	Weak = 'weak',
+	Medium = 'medium',
+	Strong = 'strong',
+	VeryStrong = 'very_strong'
+}
```
Comment (avolinsky) [inline src/sdk/models/entityTypes.ts:11]: missing newline

File: src/sdk/models/entityTypes.ts (hunk 3 above)
Comment (avolinsky) [reply src/sdk/models/entityTypes.ts:27]: Should we split into different types instead like on back-end? e.g. `PatternEntityType`, `TransformEntityType` etc? or are we going to add all properties from all types here?

**Comment by gotvino** [comment]
@bentatum What is the status of this PR since the card was merged in vnext last week?

**Comment by jpartin2** [comment]
> @bentatum What is the status of this PR since the card was merged in vnext last week?

@gotvino @bentatum I actually merged these changes into the ACT-1091 cli PR that I just committed. Maybe have a look at that commit and just close this one?

**Pull request closed without merging**